protein_lm/utils/fim_collator.py

```diff
@@ -0,0 +1,88 @@
+from transformers import DataCollatorForLanguageModeling
+from protein_lm.tokenizer.tokenizer import AptTokenizer
+from typing import Any, Callable, Dict, List, NewType, Optional, Tuple, Union
+from torch.utils.data import Dataset, DataLoader
+import torch
+import random
+from collections.abc import Mapping
+from transformers import BertTokenizer
+import numpy as np
+
+class FimDataCollator(DataCollatorForLanguageModeling):
+    def __init__(self,
+        tokenizer=AptTokenizer,     
+        min_span_len: int = 1, 
+        max_span_len: int = 10,
+        fim_freq: float = 0.5, 
+        mlm_probability=0.15,
+        mlm=False,
+        **kwargs ):
+
+        super().__init__(tokenizer=tokenizer,mlm_probability=mlm_probability, **kwargs)
+        self.fim_freq = fim_freq
+        self.tokenizer=AptTokenizer()
+        self.min_span_len=  min_span_len
+        self.max_span_len= max_span_len
+        self.mlm=mlm
+
+    def fim_transform(self, inputs):
+        max_span_len = min(self.max_span_len, inputs.shape[-1])
+        if self.min_span_len == max_span_len:
+          span_len=self.min_span_len
```

Comment:
Can you make sure you use a consistent tab size of 4 spaces, there is a mix in this file

Reply:
fixed!